src/ripple/rpc/handlers/PayChanClaim.cpp

```diff
@@ -31,8 +31,37 @@
 #include <ripple/rpc/impl/RPCHelpers.h>
 #include <ripple/rpc/impl/Tuning.h>
 
+#include <boost/optional.hpp>
+
 namespace ripple {
 
+static
+boost::optional<std::uint64_t>
```

Comment:
Since there are now two identical implementations of this function, how would you feel about putting it in `basics/StringUtilities`? That way if there are future bugs to fix they will only need to be fixed in one place.

Reply:
I considered that, but I decided they shouldn't be project-wide utilities. I agree code duplication like this is bad, but too many project-wide utility functions has a cost too. It makes finding the right utility function hard to find because the useful ones are mixed in with ones like this that are only used in two files.

If the two places were in the same subsystem, I'd look for a place in that subsystem to put them. As is, they are in `ripple/net` and `ripple/rpc`, so the util would have to live in `ripple`. 

I'm OK promoting this project-wide if you feel strongly, but I still vote for the two (small) duplicated functions.

Reply:
I don't feel strongly about it.  I have a lot of sympathy for the costs associated with too many project-wide utility functions.  I regularly thrash about trying to find where some utility function is hidden.  I also probably accidentally create duplicate code, since I have no idea that a function already exists that does what I want to do.  I'm not sure precisely what to do about either of those problems.

At any rate, my personal take is that this is not an overly specialized function, so I'd be fine with promoting it as a general facility (since it took you a couple of tries to get it to its current state of correctness).  But I'm also okay with duplicating it.